packages/back-end/src/services/auth/index.ts

```diff
@@ -118,23 +123,31 @@ export async function processJWT(
     project?: string | (string | undefined)[] | undefined,
     envs?: string[] | Set<string>
   ) => {
+    if (!req.userId || !req.organization) return false;
+
+    const userPermissions = getUserPermissions(
+      req.userId,
+      req.organization,
+      req.teams
+    );
     let checkProjects: (string | undefined)[];
     if (Array.isArray(project)) {
-      checkProjects = project.length > 0 ? project : [undefined];
+      checkProjects =
+        project.length > 0 ? project : Object.keys(userPermissions.projects);
```

Comment:
Now, instead of passing in an `[undefined]` array, if `project` is an empty array, we pass in all projects the user has project-specific permissions for.